android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java

```diff
@@ -208,110 +211,83 @@ public static boolean initSo(String libName, int version, IWXUserTrackAdapter ut
-          soName = path + STARTUPSO;
-        } else {
-          soName = path + STARTUPSOANDROID15;
-        }
+      String pkgName = WXEnvironment.getApplication().getPackageName();
+      String cacheFile = WXEnvironment.getApplication().getApplicationContext().getCacheDir().getPath();
 
-        File oldfile = new File(soName);
+      // cp weexjsb any way
+      // if android api < 16 copy libweexjst.so else copy libweexjsb.so
+      boolean pieSupport = true;
+      File newfile;
+      String startSoName = WXEnvironment.CORE_JSB_SO_NAME;
+      String startSoPath = STARTUPSO;
+      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+        pieSupport = false;
+        startSoName = WXEnvironment.CORE_JST_SO_NAME;
+        startSoPath = STARTUPSOANDROID15;
+      }
 
+      final File copyPath = _desSoCopyFile(startSoName);
+      if(!copyPath.exists()) {
+        copyPath.mkdirs();
+      }
+      newfile = new File(copyPath + startSoPath);
+      WXEnvironment.CORE_JSB_SO_PATH = newfile.getAbsolutePath();
+      String jsb = WXEnvironment.getDefaultSettingValue(startSoName, "-1");
+      if(newfile.exists() && TextUtils.equals(WXEnvironment.getAppVersionName(), jsb)) {
+        // no update so skip copy
+        return;
+      }
 
-        if(!oldfile.exists()) {
-          try {
-            String weexjsb = ((PathClassLoader) (WXSoInstallMgrSdk.class.getClassLoader())).findLibrary(startSoName);
-            oldfile = new File(weexjsb);
-          } catch (Throwable throwable) {
-            // do nothing
-          }
+      String path = "/data/data/" + pkgName + "/lib";
+      if (cacheFile != null && cacheFile.indexOf("/cache") > 0) {
+        path = cacheFile.replace("/cache", "/lib");
+      }
 
-        }
+      String soName;
+      if (pieSupport) {
+        soName = path + STARTUPSO;
+      } else {
+        soName = path + STARTUPSOANDROID15;
+      }
 
-        if (oldfile.exists()) {
-          WXFileUtils.copyFile(oldfile, newfile);
-        } else {
-          WXEnvironment.extractSo();
-        }
+      File oldfile = new File(soName);
 
-        Closeable w = null;
+      if (!oldfile.exists()) {
         try {
-          if(!versionFile.exists())
-            versionFile.createNewFile();
-          FileWriter fileWriter = new FileWriter(versionFile);
-          w = fileWriter;
-          fileWriter.write(String.valueOf(WXEnvironment.CORE_JSB_SO_VERSION));
-          fileWriter.flush();
-        } catch (Exception e ) {
+          String weexjsb = ((PathClassLoader) (WXSoInstallMgrSdk.class.getClassLoader())).findLibrary(startSoName);
+          oldfile = new File(weexjsb);
+        } catch (Throwable throwable) {
           // do nothing
-        } finally {
-          if(w != null)
-            w.close();
         }
+      }
+
+      if(!oldfile.exists()) {
+        WXEnvironment.extractSo();
+        oldfile = new File(copyPath, STARTUPSO);
+      }
 
+      if (oldfile.exists()) {
```

Comment:
The old code for line 287 is a `else` case. You are changing the behaviour in new code.